app/javascript/react/screens/App/Overview/components/ModalWizard/ModalWizard.js

```diff
@@ -14,10 +16,23 @@ const ModalWizard = props => {
     activeStep,
     numSteps,
     goToStep,
-    children
+    children,
+    formContainer
   } = props;
   const onFirstStep = activeStepIndex === 0;
   const onFinalStep = activeStepIndex === numSteps - 1;
+
+  const formHasErrors = () =>
+    find(formContainer, form => form.syncErrors) !== undefined;
+
+  const disableNextStep = formHasErrors();
+
+  const onCancel = () => {
+    // Delete/Destroy all wizard forms here
+    delete formContainer.mappingWizardGeneralStep;
```

Comment:
Calling `delete` on a property of one of your `props`? hmm... we should never be mutating our props. This should probably instead be some kind of call to a callback from higher up that changes the formContainer prop from above. @priley86 what do you think here?

Reply:
I think there should be a redux action for form cancellation, this should just dispatch that. A reducer should handle the mutation of formContainer, and not by mutating it directly but by constructing an copy of it with the changes.  i.e., in the reducer:

```
const newFormContainer = { ...oldFormContainer, mappingWizardGeneralStep: null };
```

Or if you want to use one of my new helpers, `filterKeys`:

```
import { filterKeys } from 'patternfly-react';
const newFormContainer = filterKeys(oldFormContainer, key => key !== 'mappingWizardGeneralStep');
```

Alternatively, it looks like you might get a `resetForm()` prop injected that you can use instead: https://redux-form.com/6.0.0-alpha.4/docs/faq/howtoclear.md/

Reply:
I went through this exercise with `resetForm()` but had some difficulties making it work

Will give it a try again and let you know what I find.

Reply:
I went with Approach D in https://redux-form.com/6.0.0-alpha.4/docs/faq/howtoclear.md/, which seems to be working great.

Addressed in f44fa53

Reply:
after discussing w/ you guys just now... i think we can just inject the form inputs into the MappingWizard via the same redux-form store keys... (so inputs can be read there as well as the sub form)... then when all forms become valid....we can construct the POST body for the mapping in MappingWizard. MappingWizard becomes the primary state manager I suppose.